### PR TITLE
Enable elastic input width for Layernorm layer

### PR DESCRIPTION
**Pull request by jpablomch**
### Changes

Enable elastic input width for Layernorm layer.

### Reason for changes

Required for transformer-based super-networks.

### Related tickets

N/A

### Tests

Added test for layernorm layer.

### CODE REVIEW

**Comment by jpablomch** [comment]
@ljaljushkin

**Comment by ljaljushkin** [comment]
@jpablomch please fix pylint and pytorch precommit

**Comment by ljaljushkin** [comment]
@daniil-lyakhov please take a look at the pruning part

**Comment by daniil-lyakhov** [comment]
Thank you for your PR, @jpablomch! But unfortunately it looks unfinished from filter pruning algorithm perspective.
LayerNorms operation should have filter pruning prehook like BatchNorm/GroupNorm operation to actually zero channels in a feature map. 
To do this one need to 
 * update `types_to_apply_mask` here
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/pruning/base_algo.py#L143-L146
* update tests in the file tests/torch/pruning/filter_pruning/test_algo.py and check that parameters of layernorm is pruned correctly (check that feeded zeros isn't changed to non zero values by layernorm)

**Comment by ljaljushkin** [comment]
> * update tests in the file `tests/torch/pruning/filter_pruning/test_algo.py` and check that parameters of layernorm is pruned correctly (check that feeded zeros isn't changed to non zero values by layernorm)

Thanks for the review Daniil!
Could you please elaborate, what do you mean in the second bullet?
Do you mean adding LayerNorm to `BigPruningTestModel` and checking mask and module output in the `test_pruning_masks_applying_correctness`?

**Comment by daniil-lyakhov** [comment]
> > * update tests in the file `tests/torch/pruning/filter_pruning/test_algo.py` and check that parameters of layernorm is pruned correctly (check that feeded zeros isn't changed to non zero values by layernorm)
> 
> Thanks for the review Daniil! Could you please elaborate, what do you mean in the second bullet? Do you mean adding LayerNorm to `BigPruningTestModel` and checking mask and module output in the `test_pruning_masks_applying_correctness`?

Yes, and update `test_pruning_masks_correctness` test as well

**Review by ljaljushkin — CHANGES_REQUESTED** [review]
please fix pylint and pytorch precommit

**Comment by daniil-lyakhov** [comment]
@jpablomch, please be aware of following minor pylint errors:
```
************* Module tests.torch.nas.test_elastic_width
06:13:28  tests/torch/nas/test_elastic_width.py:50:0: C0301: Line too long (147/120) (line-too-long)
06:13:28  ************* Module tests.torch.pruning.filter_pruning.test_algo
06:13:28  tests/torch/pruning/filter_pruning/test_algo.py:269:0: R0915: Too many statements (61/60) (too-many-statements)
```

**Review by daniil-lyakhov — APPROVED** [review]
Check filter pruning changes locally, LGTM!

**Review by ljaljushkin — APPROVED** [review]
LGTM
Will merge it as soon as PR will be unblocked for merging: still pylint is failing, @jpablomch please fix.

**Comment by jpablomch** [comment]
> @jpablomch, please be aware of following minor pylint errors:
> 
> ```
> ************* Module tests.torch.nas.test_elastic_width
> 06:13:28  tests/torch/nas/test_elastic_width.py:50:0: C0301: Line too long (147/120) (line-too-long)
> 06:13:28  ************* Module tests.torch.pruning.filter_pruning.test_algo
> 06:13:28  tests/torch/pruning/filter_pruning/test_algo.py:269:0: R0915: Too many statements (61/60) (too-many-statements)
> ```

Fixed. Thanks!
